lib/cli.ex

```diff
@@ -22,30 +22,33 @@ defmodule Onigumo.CLI do
 
             case switches do
               [] -> module.main(working_dir)
-              _ -> usage_message()
+              _ -> usage_message("unknown options")
             end
 
           :error ->
-            usage_message()
+            usage_message("unknown positional argument")
         end
 
       {_, _, [_ | _]} ->
```

Comment:
To be specific below, we’d need to name the third element.

```suggestion
      {_, _, invalid} when length(invalid) > 0 ->
```

Reply:
```Compiling 1 file (.ex)
    warning: do not use "length(invalid) > 0" to check if a list is not empty since length always traverses the whole list. Prefer to pattern match on a non-empty list, such as [_ | _], or use "invalid != []" as a guard```